src/components/WalletList.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import './WalletList.css';
+
+type WalletEntryParams = {
+  name: string;
+  uuid: string;
+  rdns: string;
+  icon: string;
+  extensionId?: string;
+};
+
+export type WalletMapEntry = {
+  params: WalletEntryParams;
+  eventName: string;
+};
+
+type WalletListProps = {
+  wallets: Record<string, WalletMapEntry>;
+  handleClick: (extensionId: string) => void;
+};
+
+const WalletList: React.FC<WalletListProps> = ({ wallets, handleClick }) => {
+  const walletEntries = Object.entries(wallets);
+
+  if (!walletEntries.length) {
+    return null;
+  }
+
+  return (
+    <div className="wallet-list">
+      {walletEntries.map(([key, wallet]) => (
+        <div key={key} className="wallet-card">
+          <img
+            src={wallet.params.icon}
+            alt={`${wallet.params.name} icon`}
+            className="wallet-icon"
+          />
+          <div className="wallet-info">
+            <h3 className="wallet-name">{wallet.params.name}</h3>
+            <p className="wallet-uuid">UUID: {wallet.params.uuid}</p>
+            <p className="wallet-rdns">RDNS: {wallet.params.rdns}</p>
+            {wallet.params.extensionId && (
+              <p className="wallet-extension-id">
+                Extension ID: {wallet.params.extensionId}
+              </p>
+            )}
+          </div>
+          <button
+            disabled={Boolean(!wallet.params.extensionId)}
+            onClick={() => handleClick(String(wallet.params.extensionId))}
```

Comment:
is this this `String()` necessary?

Reply:
resolved in [de81e](https://github.com/MetaMask/test-dapp-multichain/pull/22/commits/de81e7e392022e41ac8edd4066a44826540fb4d5)